Source/WebCore/animation/WebAnimation.cpp

```diff
@@ -363,9 +363,21 @@ void WebAnimation::effectTargetDidChange(const std::optional<const Styleable>& p
     InspectorInstrumentation::didChangeWebAnimationEffectTarget(*this);
 }
 
+bool WebAnimation::isTimeValid(const std::optional<CSSNumberishTime>& time) const
+{
+    // https://drafts.csswg.org/web-animations-2/#validating-a-css-numberish-time
+    if (time && !time->isValid())
+        return false;
+    if (m_timeline && m_timeline->isProgressBased() && time && time->time())
+        return false;
+    if ((!m_timeline || m_timeline->isMonotonic()) && time && time->percentage())
```

Comment:
Why does the spec express the last check as negation? 

e.g. "Is not a duration" instead of "is a percentage" 

Is there a specific reason? or is spec language just unnecessarily complicated?

Reply:
Good question, but I don't know.